main/document/document.php

```diff
@@ -1110,68 +1137,96 @@ function convertModal (id, format) {
 
         // Security fix: make sure they can't move files that are not in the document table
         if (!empty($document_to_move)) {
-            $real_path_target = $base_work_dir.$moveTo.'/'.basename($document_to_move['path']);
-            $fileExist = false;
-            if (file_exists($real_path_target)) {
-                $fileExist = true;
-            }
-            if (move($base_work_dir.$document_to_move['path'], $base_work_dir.$moveTo)) {
-                DocumentManager::updateDbInfo(
-                    'update',
-                    $document_to_move['path'],
-                    $moveTo.'/'.basename($document_to_move['path'])
-                );
-
-                //update database item property
-                $doc_id = $_POST['move_file'];
-                if (is_dir($real_path_target)) {
-                    api_item_property_update(
-                        $courseInfo,
-                        TOOL_DOCUMENT,
-                        $doc_id,
-                        'FolderMoved',
-                        api_get_user_id(),
-                        $group_properties,
-                        null,
-                        null,
-                        null,
-                        $sessionId
-                    );
-                    Display::addFlash(Display::return_message(get_lang('DirMv'), 'confirmation'));
-                } elseif (is_file($real_path_target)) {
-                    api_item_property_update(
-                        $courseInfo,
-                        TOOL_DOCUMENT,
-                        $doc_id,
-                        'DocumentMoved',
-                        api_get_user_id(),
-                        $group_properties,
-                        null,
-                        null,
-                        null,
-                        $sessionId
+            if ($document_to_move['filetype'] == 'link') {
+                $real_path_target = $base_work_dir.$_POST['move_to'].'/';
+                if (!DocumentManager::cloudLinkExists($_course, $_POST['move_to'], $document_to_move['comment'])) {
+                    $doc_id = $_POST['move_file'];
+            
+                    DocumentManager::updateDBInfoCloudLink($document_to_move['path'], $_POST['move_to'].'/', $doc_id);
```

Comment:
Please make 100% sure that the $_POST['move_file'] (int?) and $_POST['move_to'] (string?) values is checked against XSS, CRLF or directory traversal attacks before using them into another string.

I'll continue checking that below.